rle/rleplus_reader.go

```diff
@@ -1,9 +1,57 @@
 package rlepluslazy
 
 import (
+	"math/bits"
+
 	"golang.org/x/xerrors"
 )
 
+type decodeInfo struct {
+	length byte // length of the run
+	i      byte // i+1 is number of repeats of above run lengths
+	n      byte // number of bits to read
```

Comment:
Followup: nibbles?

Reply:
Sorry, I don't follow.

Reply:
If you mean packing it in 4 bit fragments, possible (I have a copy locally) but it is only 0.5% faster so not really worht the complexity.

Reply:
Yes. Oh well... I guess we'll just have to live with 40%. :man_shrugging: